Controller/Cards/DeleteAction.php

```diff
@@ -0,0 +1,128 @@
+<?php
+namespace Omise\Payment\Controller\Cards;
+
+use Magento\Framework\App\Request\Http;
+use Magento\Framework\Exception\NotFoundException;
+
+class DeleteAction extends \Magento\Framework\App\Action\Action
+{
+    const WRONG_REQUEST = 1;
+
+    const WRONG_TOKEN = 2;
+
+    const ACTION_EXCEPTION = 3;
+
+    /**
+     * @var array
+     */
+    private $errorsMap = [];
+
+    /**
+     * @var Magento\Customer\Model\Session
+     */
+    private $customerSession;
+
+    /**
+     * @var Omise\Payment\Model\Customer
+     */
+    private $customer;
+
+    /**
+     * @param Context  $context
+     * @param Session  $customerSession
+     * @param Customer $customer
+     */
+    public function __construct(
+        \Magento\Framework\App\Action\Context $context,
+        \Magento\Customer\Model\Session       $customerSession,
+        \Omise\Payment\Model\Customer         $customer
+    ) {
+        parent::__construct($context, $customerSession);
+        $this->customerSession = $customerSession;
+        $this->customer        = $customer;
+
+        $this->errorsMap = [
+            self::WRONG_TOKEN      => __('No token found.'),
+            self::WRONG_REQUEST    => __('Wrong request.'),
+            self::ACTION_EXCEPTION => __('Deletion failure. Please try again.'),
+        ];
+    }
+
+    /**
+     * Dispatch request
+     *
+     * @return Magento\Framework\Controller\ResultInterface|Magento\Framework\App\ResponseInterface
+     * @throws NotFoundException
```

Comment:
Just wonder, in what case this would throw NotFoundException?

Reply:
i copied it from existing magento code, You are right, will remove that comment.